pkg/apiserver/endpoints/filters/profiling_labels.go

```diff
@@ -0,0 +1,27 @@
+package filters
+
+import (
+	"context"
+	"net/http"
+	"runtime/pprof"
+
+	"k8s.io/apiserver/pkg/endpoints/request"
+
+	grafanarequest "github.com/grafana/grafana/pkg/services/apiserver/endpoints/request"
+)
+
+func WithProfilingLabels(handler http.Handler) http.Handler {
```

Comment:
@toddtreece what's the recommendation here, allow middlewares to be provided to or a func to return which labels to be used in builder/helper.go?

Or should the `grafanarequest.ParseNamespace` be moved here?

Reply:
Or re-implement the parsing in the middleware?

Reply:
the logic to extract a stackID is fairly simple, you can copy that here if you want to